kernel/src/locking/spinlock.rs

```diff
@@ -93,12 +107,14 @@ pub struct SpinLock<T> {
     /// protected data. That is, it allows the data to be accessed/modified
     /// while enforcing the locking mechanism.
     data: UnsafeCell<T>,
+    /// Use generic type I in the struct without consuming space.
+    phantom: PhantomData<I>,
```

Comment:
This makes the lock `!Send` and `!Sync` if `I` is `!Send` and `!Sync`.

Try this instead:
 ```suggestion
     phantom: PhantomData<fn(I)>,
```

Reply:
No, it doesn't because we have explicit `Send` and `Sync` impls anyway. Please disregard my previous comment.

Reply:
Okay, changed for `RawSpinLock` and `RawRWLock`.